adapters/arpae.js

```diff
@@ -0,0 +1,292 @@
+'use strict';
+
+import { REQUEST_TIMEOUT } from '../lib/constants';
+import { default as baseRequest } from 'request';
+import { default as moment } from 'moment-timezone';
+import { acceptableParameters, convertUnits } from '../lib/utils';
+const request = baseRequest.defaults({timeout: REQUEST_TIMEOUT});
+
+exports.name = 'arpae';
+
+const ckanResourceID = 'a1c46cfe-46e5-44b4-9231-7d9260a38e68';
+const searchUrl = 'https://dati.arpae.it/api/action/datastore_search';
+
+exports.fetchData = function (source, cb) {
+  const yesterday = moment.tz(source.timezone).subtract(1, 'days').format('Y-M-D');
+  // not doing paging right now, so set a limit
+  const limit = 100 * acceptableParameters.length;
+  const queryUrl = `${searchUrl}?resource_id=${ckanResourceID}&filters={"reftime": "${yesterday}"}&limit=${limit}`;
+
+  request(queryUrl, (err, res, body) => {
+    if (err || res.statusCode !== 200) {
+      return cb(err || res);
+    }
+    const data = JSON.parse(body);
+    const records = data['result']['records'];
+    let measurements = [];
+    records.forEach((record) => {
+      const parameter = parameters[record.variable_id].parameter;
+      if ((acceptableParameters.indexOf(parameter) === -1) ||
+          (!locations[record.station_id])) {
+        // unexpected parameter or
+        // location detail unknown, skip
+        return;
+      }
+      const unit = parameters[record.variable_id].unit;
+      const location = locations[record.station_id].station;
+      const city = locations[record.station_id].comune;
+      const coordinates = locations[record.station_id].coordinates;
+
+      const mDate = moment.tz(record.reftime, source.timezone);
+      measurements.push({
+        value: Number(record.value),
+        unit: unit,
+        parameter: parameter,
+        date: {
+          utc: mDate.toDate(),
+          local: mDate.format()
+        },
+        country: source.country,
+        city: city,
+        location: location,
+        coordinates: coordinates,
+        attribution: [
```

Comment:
Can we get averagingPeriod added in here somewhere? Is it 1 hr?

Reply:
Oh right.